.github/workflows/helm-build-push.yml

```diff
@@ -0,0 +1,53 @@
+name: Helm chart release on tag
+
+on:
+  workflow_call:
+    charts:
+      type: str
+      description: "What Charts should be updated? Can be comma separated list."
+
+jobs:
+  release-helm-chart:
+    name: Release helm chart
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        chart:
+          [ split(${{ github.event.inputs.charts }}, ',') ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get version from tag
+        shell: bash
+        run: |
+          vtag='${{github.ref_name}}'
+          echo "TAG=${vtag:1}" >> $GITHUB_ENV
+
+      - name: Run helm version upgrade
+        uses: greenbone/actions/helm-version-upgrade@v2
+        with:
+          chart-path: ${{ github.workspace }}/charts/${{ matrix.chart }}
+          chart-version: ${{ env.TAG }}
+
+      - name: Print Chart.yaml
+        run: |
+          cat '${{ github.workspace }}/charts/${{ matrix.chart }}/Chart.yaml'
+
+      - name: Upload to github registry
+        uses: greenbone/actions/helm-build-push@v2
+        with:
+          chart-name: ${{ matrix.chart }}
+          registry: ${{ vars.IMAGE_REGISTRY }}
+          registry-subpath: helm-charts/
+          registry-user: ${{ secrets.GREENBONE_BOT }}
```

Comment:
IMHO secrets must be handled specially in a reusable workflow https://docs.github.com/en/actions/using-workflows/reusing-workflows#creating-a-reusable-workflow